SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -33,6 +34,143 @@ DECLARE_CYCLE_STAT(TEXT("AddUserDefinedQueries"), STAT_InterestFactoryAddUserDef
+	TArray<Worker_ComponentSetId> ComponentSets;
+	TArray<Worker_EntityId_Key> Entities;
+
+	void DebugOutput(const FString& DiffType) const
+	{
+		// Minimal output
+		if (Components.Find(SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID) != INDEX_NONE)
+		{
+			UE_LOG(LogTemp, Warning, TEXT("Interest diff: client auth interest"));
+		}
+		else
+		{
+			UE_LOG(LogTemp, Warning, TEXT("Interest diff: client non auth interest"));
+		}
+
+		/*
+		// Verbose output
+		FString ComponentsString;
+		for (const auto& Component : Components)
+		{
+			ComponentsString += FString::Format(TEXT("{0} "), { Component });
+		}
+
+		UE_LOG(LogTemp, Warning, TEXT("Interest diff: components %s"), *ComponentsString);
+
+		FString ComponentSetsString;
+		for (const auto& ComponentSet : ComponentSets)
+		{
+			ComponentSetsString += FString::Format(TEXT("{0} "), { ComponentSet });
+		}
+
+		UE_LOG(LogTemp, Warning, TEXT("Interest diff: component sets %s"), *ComponentSetsString);
+		*/
+
+		FString EntitiesString;
+		for (const auto& EntityId : Entities)
+		{
+			EntitiesString += FString::Format(TEXT("{0} "), { EntityId });
+		}
+
+		UE_LOG(LogTemp, Warning, TEXT("Interest diff: %s %s"), *DiffType, *EntitiesString);
+	}
+};
+
+struct ChangeInterestRequest
+{
+	Worker_EntityId SystemEntityId;
+	TArray<MyQuery> QueriesToAdd;
+	TArray<MyQuery> QueriesToRemove;
+	bool bOverwrite = false;
+
+	void Clear()
+	{
+		QueriesToAdd.Empty();
+		QueriesToRemove.Empty();
+		bOverwrite = false;
+	}
+
+	bool IsEmpty() const { return QueriesToAdd.Num() == 0 && QueriesToRemove.Num() == 0; }
+
+	void DebugOutput() const
+	{
+		UE_LOG(LogTemp, Warning, TEXT("Interest diff: client entity id %lld"), SystemEntityId);
+
+		for (const auto& Query : QueriesToAdd)
+		{
+			Query.DebugOutput(TEXT("ADD"));
+		}
+
+		for (const auto& Query : QueriesToRemove)
+		{
+			Query.DebugOutput(TEXT("REMOVE"));
+		}
+
+		UE_LOG(LogTemp, Warning, TEXT("Interest diff: overwrite %d"), bOverwrite);
+	}
+
+	void CreateRequest(Worker_CommandRequest& CommandRequest) const
+	{
+		/*
+			type ResultType {
+			  list<uint32> components = 1;
+			  list<uint32> component_sets = 2;
+			}
+
+			type Query {
+			  ResultType result_type = 1;
+			  list<int64> entities = 2;
+			}
+
+			type ChangeInterestRequest {
+			  int64 worker_entity_id = 1;
+			  list<Query> queries_to_add = 2;
+			  list<Query> queries_to_remove = 3;
+			  bool overwrite = 4;
+			}
+		*/
+
+		CommandRequest.component_id = SpatialConstants::WORKER_COMPONENT_ID;
+		CommandRequest.command_index = SpatialConstants::WORKER_CHANGE_INTEREST_COMMAND_ID;
+		CommandRequest.schema_type = Schema_CreateCommandRequest();
+
+		Schema_Object* RequestObject = Schema_GetCommandRequestObject(CommandRequest.schema_type);
+
+		Schema_AddEntityId(RequestObject, 1, SystemEntityId);
+
+		if (QueriesToAdd.Num() > 0)
+		{
+			for (const MyQuery& Query : QueriesToAdd)
+			{
+				Schema_Object* QueriesToAddObject = Schema_AddObject(RequestObject, 2);
```

Comment:
nit. pls move this down either:
- 1 line 
- or to be just above the `Schema_AddEntityIdList` line